pkg/gql/httpserver.go

```diff
@@ -90,8 +96,50 @@ func (s *Server) Start() error {
 	s.schema = &sc
 	_, s.db = heavy.CreateDBConnection()
 
-	// Set up listener
-	l, err := net.Listen("tcp", "localhost:"+cfg.Get().Gql.Port)
+	if network == "unix" {
+		if err := os.RemoveAll(address); err != nil {
+			return err
+		}
+	}
+
+	var tlsCert *tls.Certificate
+
+	certFile := cfg.Get().Gql.CertFile
+	keyFile := cfg.Get().Gql.KeyFile
+
+	if len(certFile) > 0 {
+
+		// Fetch certificate from config
+		tlsCertFromFile, err := tls.LoadX509KeyPair(certFile, keyFile)
+		if err != nil {
+			return err
+		}
+		tlsCert = &tlsCertFromFile
+	} else {
+
+		// Generate self-signed certificate
+		tlsCert, err = cryptoUtils.GenerateTLSCertificate()
+		if err != nil {
+			return err
+		}
+	}
+
+	// Define TLS configuration
+	tlsConfig := tls.Config{
+		Certificates:       []tls.Certificate{*tlsCert},
+		InsecureSkipVerify: true,
```

Comment:
`InsecureSkipVerify` concerns only the client TLS connection, right? 
Do we need it enabled here?

Reply:
Also correct, and fixed. Late we will have safe certs